source/features/set-default-repositories-type-to-sources.tsx

```diff
@@ -1,25 +1,36 @@
 import select from 'select-dom';
+import elementReady from 'element-ready';
 
 import features from '.';
 
-function init(): void {
+function addSourceTypeToLink(link: HTMLAnchorElement): void {
+	const search = new URLSearchParams(link.search);
+	search.set('type', 'source');
+	link.search = String(search);
+}
+
+function headerDropdownListener(): void {
+	addSourceTypeToLink(select<HTMLAnchorElement>('.header-nav-current-user ~ [href$="tab=repositories"]')!);
+}
+
+async function init(): Promise<void> {
 	const links = select.all<HTMLAnchorElement>([
 		// Pre "Repository refresh" layout
-		'.user-profile-nav [href$="tab=repositories"]', // "Your repositories" in header dropdown
 		'#user-links [href$="tab=repositories"]', // "Repositories" tab on user profile
 		'.orgnav > .pagehead-tabs-item:first-child', // "Repositories" tab on organization profile
 		// "Repository refresh" layout
-		'.header-nav-current-user ~ [href$="tab=repositories"]', // "Your repositories" in header dropdown
 		'[aria-label="User profile"] [href$="tab=repositories"]', // "Repositories" tab on user profile
 		'[aria-label="Organization"] .UnderlineNav-item:first-child', // "Repositories" tab on organization profile
 		'[data-hovercard-type="organization"]' // Organization name on repo header + organization list on user profile
 	]);
 
 	for (const link of links) {
-		const search = new URLSearchParams(link.search);
-		search.set('type', 'source');
-		link.search = String(search);
+		addSourceTypeToLink(link);
 	}
+
+	(await elementReady('[aria-label="View profile and more"]'))! // "Your repositories" in header dropdown
+		.closest('details')!
+		.addEventListener('toggle', headerDropdownListener, {once: true});
```

Comment:
Apparently this doesn't work sometimes, for both the features that use it. This happens when the user clicks the dropdown _before_ GitHub loads the dropdown content, so we need to listen for the content load instead.

This is easy-ish since we do in in other places: watch for the `load` event on its `include-fragment`.

If it's not already, it might be worth standardizing this "add listener function". Maybe something like:

```ts
onFragmentLoad(element: Element /*or whatever include-fragment is*/, handler: EventListener): void
```
or

```ts
onFragmentLoad(element: Element /*or whatever include-fragment is*/): Promise<EventListener>
```
cc @yakov116

Reply:
observeOneMutation but yeah this should be maid as its own function since not always will we need to wait for it to load since it may be there already.

Reply:
> observeOneMutation

No, it's all event-based: https://github.com/sindresorhus/refined-github/blob/4a064058d5889a5052de0abdfab089c6894d7266/source/github-events/on-new-comments.ts#L55

Reply:
> This file is almost exactly what we need, except it's specific to PR files rather than generic to `include-fragment`

So the idea would be to make `onPrFileLoad()` more generic and use it in this feature?